arrow-core-data/src/main/kotlin/arrow/core/Option.kt

```diff
@@ -441,8 +616,20 @@ sealed class Option<out A> : OptionOf<A> {
   fun <B, R> map2(fb: Kind<ForOption, B>, f: (Tuple2<A, B>) -> R): Option<R> =
     flatMap { a: A -> fb.fix().map { b -> f(a toT b) } }
 
+  @JvmName("filterMapOption")
+  @Deprecated(
+    "Use the filterMap method that receives a function from A to nullable instead",
+    ReplaceWith(
+      "this.filterMap(f.andThen { it.orNull() })",
+      "arrow.core.andThen"
+    ),
+    level = DeprecationLevel.WARNING
+  )
   fun <B> filterMap(f: (A) -> Option<B>): Option<B> =
-    flatMap { a -> f(a).fold({ empty<B>() }, { just(it) }) }
+    flatMap(f)
+
+  inline fun <B> filterMap(f: (A) -> B?): Option<B> =
```

Comment:
Duplicate with `mapNonNull`, should me favor `mapNotNull` instead? We do that for `Iterable<A>` & Collections where we favor `kotlin.collections.mapNotNull` over providing our own `filterMap` function.

Reply:
I believe so, we should use mapNotNull instead of filterMap across the board

Reply:
Ok, I'll do that.